index.js

```diff
@@ -40,7 +40,8 @@ module.exports = function(app) {
    */
 
   app.mixin('use', function(fn) {
-    var plugin = fn.call(this, this);
+    var thisArg = (typeof this === 'function' ? this.__proto__ : this);
```

Comment:
we should probably also do `&& __proto__ in this`, just to be sure. probably make it an if statement instead of ternary.